go/signedexchange/cmd/dump-signedexchange/main.go

```diff
@@ -1,53 +1,118 @@
 package main
 
 import (
+	"errors"
 	"flag"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"log"
+	"net/http"
 	"os"
+	"strings"
 	"time"
 
 	"github.com/WICG/webpackage/go/signedexchange"
+	"github.com/WICG/webpackage/go/signedexchange/version"
 )
 
+var latestVersion = string(version.AllVersions[len(version.AllVersions)-1])
+
 var (
-	flagInput     = flag.String("i", "", "Signed-exchange input file")
-	flagSignature = flag.Bool("signature", false, "Print only signature value")
-	flagVerify    = flag.Bool("verify", false, "Perform signature verification")
-	flagCert      = flag.String("cert", "", "Certificate CBOR file. If specified, used instead of fetching from signature's cert-url")
+	flagCert          = flag.String("cert", "", "Certificate CBOR file. If specified, used instead of fetching from signature's cert-url")
+	flagHeaders       = flag.Bool("headers", true, "Print headers")
+	flagFilename      = flag.String("i", "", "Signed-exchange input file")
+	flagPayload       = flag.Bool("payload", true, "Print payload")
+	flagSignature     = flag.Bool("signature", false, "Prints only signature value")
+	flagURI           = flag.String("uri", "", "Signed-exchange uri")
+	flagVerify        = flag.Bool("verify", false, "Perform signature verification")
+	flagVersion       = flag.String("version", latestVersion, "Signed exchange version")
+	flagRequestHeader = flag.String("requestHeader", "", "Request header for signed exchange")
 )
 
+func getAcceptHeader(value string) (string, error) {
+	v, parsed := version.Parse(value)
+	if parsed {
+		return fmt.Sprintf("application/signed-exchange;v=%s", v[1:]), nil
```

Comment:
Would you factor this out as `func (v Version) MimeType() string` in `version.go`?